File: src/com/l2kt/gameserver/skills/basefuncs/FuncSubDiv.java
package com.l2kt.gameserver.skills.basefuncs;

import com.l2kt.gameserver.skills.Env;
import com.l2kt.gameserver.skills.Stats;

public class FuncSubDiv extends Func
{
	public FuncSubDiv(Stats pStat, int pOrder, Object owner, Lambda lambda)
	{
		super(pStat, pOrder, owner, lambda);
	}
	
	@Override
	public void calc(Env env)
	{
		if (cond == null || cond.test(env))
			env.divValue(1 - (_lambda.calc(env) / 100));
	}
}

File: src/com/l2kt/gameserver/skills/basefuncs/FuncSubDiv.kt
package com.l2kt.gameserver.skills.basefuncs

import com.l2kt.gameserver.skills.Env
import com.l2kt.gameserver.skills.Stats

class FuncSubDiv(pStat: Stats, pOrder: Int, owner: Any, lambda: Lambda) : Func(pStat, pOrder, owner, lambda) {

    override fun calc(env: Env) {
        if (cond == null || cond?.test(env) == true)
            env.divValue(1 - _lambda!!.calc(env) / 100)
    }
}